dev/release/release-build.sh

```diff
@@ -225,9 +233,21 @@ function checkout_code {
     rm -rf $RELEASE_WORK_DIR
     mkdir $RELEASE_WORK_DIR
     cd $RELEASE_WORK_DIR
-    git clone https://git-wip-us.apache.org/repos/asf/incubator-systemml.git
+    if [[ "$GIT_BRANCH" ]]; then
+        git clone -b $GIT_BRANCH https://git-wip-us.apache.org/repos/asf/incubator-systemml.git
```

Comment:
one or the other is sufficient

but, none of your changes should be required if you invoke the script with the `--GIT_REF="branch-name"` parameter